### ISSUE TITLE
Feature: Auto-Moderation

### ISSUE THREAD

**Issue by ArcaneDisgea** [poieten-oikos/venat#2]
For the sake of clarification within this issue "admin-defined" refers to a discord admin with bot access privileges.

- Auto Ban/Kick on ${message || term}
- Auto Ban/Kick on ${spam || raid}
- Anti-Phish 
    - Slightly ties into the above spam/raid kicks/bans [this bot seems to be able to pull a scam domain list from discord and use it to ban/kick/delete.](https://github.com/Benricheson101/anti-phishing-bot/blob/main/services/bot/lib/services/domainFetcher.ts)
    - It also has [other](https://github.com/Benricheson101/anti-phishing-bot/tree/main/services/abusive-user-checker) interesting tools that we can perhaps borrow, such as storing hashes of avatars of known scam accounts and comparing users to it to pre-emptively stop scam spam.
- Auto-Slowmode?
    - Not sure if this is something that anyone would need but perhaps if the message creation rate in a certain channel over x minutes is too high it can automatically set a slowmode on the channel that is admin defined.
- Auto-Remove New users
    - Automatically kicks users where their account is newer than an admin defined value.

This is a draft of features, not necessarily complete. Just trying to get all my thoughts down.

**Comment by kalilistic** [comment]
Maybe auto-delete messages with certain words.

**Comment by ArcaneDisgea** [comment]
> Maybe auto-delete messages with certain words.

Yeah I mentally included that in the first one but didn't actually write it.

**Comment by ArcaneDisgea** [comment]
Edited with the following. (Really wish the webhook could detect edits and post them.)
- [ ] Auto Kick if no roles in x admin-defined days
- [ ] Join Gate
    - [ ] In-Browser questionnaire admin-defined - ONLY MULTIPLE CHOICE?
    - [ ] Captcha - Optional or as the only source of verification
    
Adding this to automod despite being a xiv feature because I feel it fits here better
- [ ] Lodestone Bio Verification

**Comment by kalilistic** [comment]
We could consider converting some of these bigger issues to discussions (I just realized there's a button for that). Some of these ideas might be done soon but others might be a bit - so could end up as different tickets.

**Comment by ArcaneDisgea** [comment]
Yeah I can also add some of these that there's really no discussion for to the kanban. Like "Auto Ban/Kick/Delete message on ${message || term}" is pretty bog standard not really something that needs much discussion I think. Unless we want to debate implementation details.
